presto-parser/src/main/java/com/facebook/presto/sql/parser/AstBuilder.java

```diff
@@ -592,6 +593,52 @@ public Node visitResetSession(SqlBaseParser.ResetSessionContext context)
         return new ResetSession(getLocation(context), getQualifiedName(context.qualifiedName()));
     }
 
+    @Override
+    public Node visitGrant(SqlBaseParser.GrantContext context)
+    {
+        String grantee;
+        if (context.PUBLIC() != null) {
+            grantee = "PUBLIC";
+        }
+        else {
+            grantee = context.grantee.getText();
+        }
+
+        List<String> privileges;
+        if (context.ALL() != null && context.PRIVILEGES() != null) {
+            //List all privileges  declared in 'Privilege' enum in SPI
+            privileges = ImmutableList.of("SELECT", "DELETE", "INSERT");
```

Comment:
What we meant was that instead of expanding the list of privileges from ALL -> [SELECT, INSERT, DELETE] in the parser, the AST node should encode the fact that "all" was specified. The analyzer should then do any necessary expansions.

Going from SQL -> AST -> SQL should be able to roundtrip as much as possible. In the current implementation, the query `grant all privileges on foo to testAdmin` would turn into `grant insert, update, delete on foo to testAdmin` if you print it back from the AST.